.github/workflows/sync-channels.yaml

```diff
@@ -3,6 +3,8 @@ on:
   schedule:
     # The nightly channel is usually updated at 01:00:00 UTC.
     - cron: '0 2 * * *' # *-*-* 02:00:00 UTC
+    # The stable channel is usually updated before 17:00:00 UTC
+    - cron: '0 17 * * *' # *-*-* 17:00:00 UTC
```

Comment:
Do you have any references or evidences? Seems they do not mention anything about the scheduled time in https://forge.rust-lang.org/release/process.html#release-day-thursday.

Also it seems to be always on Thursday.

Reply:
This is based on observations and looking at rust release calendar: https://calendar.google.com/calendar/ical/mozilla.com_ts4qudb88i0tbjef8rche4a6v4@group.calendar.google.com/public/basic.ics, and because stable release is done every 6 weeks, this means it happens every Thursday.

Reply:
> This is based on observations and looking at rust release calendar:

That link shows an event on UTC 1700-1800, is it the time they click the release button or only started the CI jobs?

I checked the last-modified time of the channel TOML, which seems to be fairly random:

```console
$ curl --silent https://static.rust-lang.org/dist/channel-rust-1.{71..78}.toml --head | rg last-modified
last-modified: Thu, 03 Aug 2023 18:06:55 GMT
last-modified: Tue, 19 Sep 2023 14:02:18 GMT
last-modified: Thu, 05 Oct 2023 16:06:42 GMT
last-modified: Thu, 07 Dec 2023 14:17:53 GMT
last-modified: Thu, 28 Dec 2023 16:17:53 GMT
last-modified: Thu, 08 Feb 2024 14:02:51 GMT
last-modified: Tue, 09 Apr 2024 21:35:05 GMT
last-modified: Thu, 02 May 2024 12:34:24 GMT
```

Or is there other historical records about the precise time of releases?

Reply:
I was checking github releases: https://github.com/rust-lang/rust/releases

Reply:
> That link shows an event on UTC 1700-1800, is it the time they click the release button or only started the CI jobs?

Builds happen way prior that's just target time that all work is pushed through AFAIK.

Reply:
> I checked the last-modified time of the channel TOML, which seems to be fairly random

dot releases are more chaotic as they are done on as needed basis (so they are outlines).

Reply:
> > I checked the last-modified time of the channel TOML, which seems to be fairly random
> 
> dot releases are more chaotic as they are done on as needed basis (so they are outlines).

The point zero releases are not as stable as you mentioned either.

```rust
$ curl --silent https://static.rust-lang.org/dist/channel-rust-1.{71..78}.0.toml --head | rg last-modified
last-modified: Thu, 13 Jul 2023 13:59:35 GMT
last-modified: Thu, 24 Aug 2023 13:42:47 GMT
last-modified: Thu, 05 Oct 2023 16:06:42 GMT
last-modified: Thu, 16 Nov 2023 13:45:13 GMT
last-modified: Thu, 28 Dec 2023 16:17:50 GMT
last-modified: Thu, 08 Feb 2024 14:02:51 GMT
last-modified: Thu, 21 Mar 2024 12:53:00 GMT
last-modified: Thu, 02 May 2024 12:34:25 GMT
```

Reply:
But they are all before 17:00. Sometimes there are also last minute fixes to rc version.